tests/testthat/test-population.R

```diff
@@ -0,0 +1,49 @@
+context("Population")
+populationFileName <- getTestDataFilePath("pop_10.csv")
+values <- c(1:10) * 2.5
+
+
+test_that("It can return the expected number of individual in the population", {
+  population <- loadPopulation(populationFileName)
+  expect_equal(population$count, 10)
+})
+
+test_that("It returns whether variability is defined for a parameter path", {
+  population <- loadPopulation(populationFileName)
+  expect_true(population$has("Organism|VenousBlood|Volume [l]"))
+  expect_true(population$has("Organism|VenousBlood|Volume"))
+  expect_false(population$has("NOPE"))
+})
+
+
+test_that("It can add user defined variability using a new parameter path", {
+  population <- loadPopulation(populationFileName)
+  parameterPath <- "Organism|MyParameter"
+  population$setValues(parameterPath, values)
+  expect_true(population$has(parameterPath))
+  expect_identical(population$getValues(parameterPath), values)
+})
+
+test_that("It can add user defined variability using an existing parameter path with unit", {
+  population <- loadPopulation(populationFileName)
+  parameterPath <- "Organism|VenousBlood|Volume [l]"
+  expect_true(population$has(parameterPath))
+  population$setValues(parameterPath, values)
+  expect_true(population$has(parameterPath))
+  expect_identical(population$getValues(parameterPath), values)
+})
+
+test_that("It can add user defined variability using an existing parameter path without unit", {

```

Comment:
The logic is: If the parameter was added with unit (e.g. loaded from CSV for example) it can be retrieve without the unit
However if the parameter is added WITHOUT unit (for exampel to create user defined variability, one would use the parameter$path as a parameter), then you can only retrieve the parameter without unit (because anything else does not make sense)

I think it's fairly resilient that way and allows  all features requested to be done